Changes and rework for pac4j 5

SiMiKohl opened the pull request:
- Changed pac4j to 5.3.1
- Changed to Java 11 and (optional) servlet-api 4
- Changed jersey in jersey-pac4j back to 2.35
- Copied tests to jersey228-pac4j
- "Reworked" dependencies-management
- Switched to jakarta-artifacts
- Added .fbExcludeFilterFile to .gitignore

Mainly tested with jersey 2.35, Closes #151

leleuj commented:
Thanks for this great contribution. I will review it ASAP.

leleuj commented on core/src/main/java/org/pac4j/jax/rs/pac4j/JaxRsContext.java:
```diff
@@ -44,20 +49,24 @@
 
+    private SessionStore sessionStore = null;
+
     public JaxRsContext(Providers providers, ContainerRequestContext requestContext, SessionStore sessionStore) {
```
The `sessionStore` should not be used to create the `JaxRsContext`. Both should be treated separately.

SiMiKohl replied:
This is exactly how I see it. The reason why this was initially implemented this way (not by me) is that the Context is the only implementation-specific object that is available within the filters of the core project. I have already modified this so that the SessionStore is not needed in the constructor, but is determined by the context and the config from the providers. The only other alternatives would be to make the filters implementation-specific, to register the SessionStores as providers or to rely on the SessionStores from the Config. However, this would possibly break the compatibility.

leleuj replied:
I don't know this implementation enough, but for sure, we can break the compatibility as it's a major version.
So if you have a better solution to properly separate the session store from the context, go ahead.

leleuj commented on core/src/main/java/org/pac4j/jax/rs/pac4j/JaxRsProfileManager.java:
```diff
@@ -7,27 +7,31 @@
-
-        new RequestPac4JSecurityContext((JaxRsContext) this.context).context().ifPresent(c -> c.principal = null);
-    }
+//    @Override
```
To be removed, `logout` no longer exists in the `ProfileManager`. It was too confusing.

leleuj commented on jersey228/src/main/java/org/pac4j/jax/rs/grizzly/pac4j/GrizzlyJaxRsContext.java:
```diff
@@ -22,8 +25,8 @@
 
-    public GrizzlyJaxRsContext(Providers providers, ContainerRequestContext requestContext,
-            SessionStore sessionStore, Request request) {
+    public GrizzlyJaxRsContext(Providers providers, ContainerRequestContext requestContext, SessionStore sessionStore,
```
Same feedback as for the `JaxRsContent`. The `SessionStore` should be treated separately.

leleuj reviewed (commented):
LGTM

My main concern is the separation between the `SessionStore` and the web contexts. I hope this is feasible.

leleuj commented:
@SiMiKohl I replaced the Travis build (disabled) by the Github Actions build and reverted the versions to the latest stable release. The master version is now 5.0.0-SNAPSHOT.

Can you sync with master? Thx

SiMiKohl commented:
As discussed, I have separated the SessionStore from the Context. For this I added a new provider for each implementation. Additionally I added implementation specific features which registers all needed providers by themselves. This simplifies the usage and allows an easier modification/extension. Furthermore I changed the SessionStores to Singeltons.

leleuj commented:
Excellent! Merging. Will release as soon as you confirm that everything is ok...

leleuj commented:
OK. Good. Go ahead and keep me posted when the release is needed.

SiMiKohl commented:
Just a short feedback: I´m going to update my PR next week with an update to pac4j 5.4 and an version for resteasy3 and resteasy5

jhauglid commented:
What is the status on this pull request? It would be nice to be able to move to pac4j 5.

leleuj commented:
I think we still have some pending updates coming: @SiMiKohl ?

I can release at any time.

SiMiKohl commented:
Based on https://github.com/pac4j/jax-rs-pac4j/pull/281, I currently have no other additions. I think a release (and the needed modifications in the wiki) could be performed. The only open topic is the development of versions for JakartaEE 9/Jakarta RESTful Web Services 3.0 and onwards. Unfortunately I don't have time for this at the moment, but will look into it as soon as possible.

leleuj commented:
I just released jax-rs-pac4j v5.0.0. It will show up in the central Maven repository within a few minutes/hours.

One thing we could maybe fix is the log4j dependency which has security issues in the internal Github security report. Can we remove it?

SiMiKohl commented:
Actually, there is no direct dependency to log4j. However, there are two dependencyManagement entries. The one from the parent-pom could indeed be removed. Unfortunately the entry from the pom of resteasy5-pac4j is necessary to satisfy the enforcer-plugin. Here otherwise the log4j-api would come as a provided dependency in different versions by resteasy5 itself (see below). But we could probably just update to the latest version.

```
Dependency convergence error for org.apache.logging.log4j:log4j-api:jar:2.17.1:provided paths to dependency are:
+-org.pac4j:resteasy5-pac4j:jar:5.0.1-SNAPSHOT
  +-org.jboss.resteasy:resteasy-core:jar:5.0.3.Final:provided
    +-org.jboss.logging:jboss-logging:jar:3.4.3.Final:provided
      +-org.apache.logging.log4j:log4j-api:jar:2.17.1:provided
and
+-org.pac4j:resteasy5-pac4j:jar:5.0.1-SNAPSHOT
  +-org.jboss.resteasy:resteasy-core:jar:5.0.3.Final:provided
    +-org.jboss.resteasy:resteasy-core-spi:jar:5.0.3.Final:provided
      +-org.apache.logging.log4j:log4j-api:jar:2.13.3:provided
and
+-org.pac4j:resteasy5-pac4j:jar:5.0.1-SNAPSHOT
  +-org.jboss.resteasy:resteasy-core:jar:5.0.3.Final:provided
    +-org.jboss.resteasy:resteasy-core-spi:jar:5.0.3.Final:provided
      +-org.apache.logging.log4j:log4j-core:jar:2.13.3:provided
        +-org.apache.logging.log4j:log4j-api:jar:2.13.3:provided
```

leleuj commented:
OK. Not big deal here.
